api/services/github/oauth.py

```diff
@@ -0,0 +1,61 @@
+from typing import TypedDict
+from httpx import AsyncClient
+from fastapi import status
+from urllib.parse import urlencode
+from .client import GithubApiException
+
+
+class GithubOauthRecord(TypedDict):
+    referrer: str | None  # the client's initial path prior to the authentication
+
+
+"""
+An in-memory store for keep tracking of client and server state during Github OAuth
+"""
+
+
+class GithubOauthStore:
+    def __init__(self):
+        self.__store = {}
+
+    def has(self, key: str):
+        return key in self.__store
+
+    def pop(self, key: str) -> GithubOauthRecord:
+        record = self.__store[key]
+        del self.__store[key]
+        return record
+
+    def put(self, key: str, record: GithubOauthRecord):
+        self.__store[key] = record
+
+
+class GithubOauth:
```

Comment:
Not sure why `git` didn't recognize github_oauth -> github/oauth rename, but all I did here was add `GithubOauth`.